### ISSUE TITLE
NCPA Agent doesn't properly report iptables status

### ISSUE THREAD

**Issue by ericloyd** [NagiosEnterprises/ncpa#232]
There doesn't seem to be a more specific board for NCPA, so I'm posting this here.
```
https://192.168.1.11:5693/api/service/httpd/running
```
That works fine:
```
{
  "value": {
    "returncode": 0, 
    "stdout": "OK: Service httpd is running"
  }
}
```

But change http to iptables, and it fails, even though iptables is running:
```
{
  "value": {
    "returncode": 2, 
    "stdout": "CRITICAL: Service iptables is stopped"
  }
}
```

I'm pretty sure this has to do with the fact that "service itpables status" does not return a "is running..." string, but instead returns the current results of "itpables --list"  So I'm guessing NCPA's parser doesn't see the "is running" and assumes that it "is stopped."  So can't check to see that iptables is running that way.

Also can't check via /api/processes, because it doesn't show up in the list!

Would love to switch all our NRPE stuff over to NCPA, but if we can't get accurate process/service information, we can't convert.

**Comment by jhomann** [comment]
Hey Eric, are you testing this on a production system or a test system? I have some build binaries of 2.0.0-alpha that I would like you to try if you're up for it. You'd need to use the newest check_ncpa.py from the master branch here for it to work but I believe this issue has been fixed in the next release.

**Comment by ericloyd** [comment]
It's on what we call our "production test cluster."  One (or more) of every type of Nagios box there is, including some gearman servers.

Happy to test, but won't be able to get around to it until mid-next-week.

**Comment by jhomann** [comment]
Sounds good, I will PM you the location since I'd rather not have people use these in production. Let me know anything you find. I will test this myself too to see if I can replicate it.

**Comment by jhomann** [comment]
This will probably need to either be accounted for or we will need to rely on psutils more now that it's a higher version with a lot less issues! I just tested this on Linux and it looks like iptables is indeed showing as stopped even though it is running. With the latest version in master.

I doubt you'd see it in the processes. Using `ps -ef` I don't actually see it myself, I am curious what it runs as ... Hmm according to online sources it's not a daemon so it shouldn't show up

**Comment by ccztux** [comment]
It runs as a kernel module:

```
[root@localhost]:~# lsmod |grep ip_table
ip_tables              17831  2 iptable_filter,iptable_nat
```

**Comment by ericloyd** [comment]
Indeed.  Kernel modules don't show up as processes or services, but sometimes can be treated as such.  In this case though, if it's stopped, you get "iptables: Firewall is not running" which means it can be checked to see if it's not running, but it can't be checked to see if it is running.

**Comment by jhomann** [comment]
I will definitely get this check fixed I am just waiting on getting a new Dev environment set up on Cent 6 to easily test it. Will definitely be fixed in the actual release of 2.0.0 though!

**Comment by ericloyd** [comment]
Groovy.  Thanks!

**Comment by jhomann** [comment]
This is actually also an issue with systemd but I have fixed it. On to cent 6.

**Comment by jhomann** [comment]
This is fixed for RHEL/CentOS 6 using f1c5fd0364f216b5a68cd80ac7b2a9520bee089c which is probably not a perfect solution but neither is checking `ps` if a service is running. This at least allows us to check a separate way for any modules or services that do not show up in the future.
